### ISSUE TITLE
Segmenation performance drops compares to reported in paper.

### ISSUE THREAD

**Issue by JianwenCao** [facebookresearch/dinov2#517]
I am using base model with linear head for semantic segmentation in VOC2012. I do not train myself, only downloading from the official weights. I achieved 0.738 miou in validation set (While 0.825 reported in the paper).

----------------------------------------------------------------------------------------------------
---Here is how I transform the data, the inference size is 518:
self.valid_preprocessors = [ToTensor(),
                            ResizeKeepRatio(self.config["inference_size"]),
                            CenterCrop(self.config["inference_size"]),
                            Normalize(mean=self.config["mean"], std=self.config["std"])]

----------------------------------------------------------------------------------------------------
---Here is how I forward the model, all pretrained parameters are strictly loaded:
def forward(self, x):
    ori_h, ori_w = x.shape[-2:]
    x = self.pad(x) 
    h, w = x.shape[-2:]
    fh, fw = h//self.patch_size, w//self.patch_size

    if self.config["enable_ms"]:
        x = self.backbone._get_intermediate_layers_not_chunked(x, n=4)
        x = [rearrange(i[:, 1:], "b (fh fw) c -> b c fh fw", fh=fh, fw=fw) for i in x]
        x = self.transform_inputs(x)
    else:
        x = self.backbone.forward_features(x)["x_norm_patchtokens"]
        x = rearrange(x, "b (fh fw) c -> b c fh fw", fh=fh, fw=fw)
    x = self.decode_head(x, target_size=(ori_h, ori_w))
    return x

---More detais in decode_head:
class BNHead(nn.Module):

  def __init__(self, embed_dim, num_classes):
      super().__init__()
      self.bn = nn.SyncBatchNorm(embed_dim)
      self.conv_seg = nn.Conv2d(in_channels=embed_dim, out_channels=num_classes, kernel_size=(1, 1), padding=(0, 0))

  def forward(self, x, target_size):
      x = self.bn(x)
      x = self.conv_seg(x)
      x = nn.functional.interpolate(x, target_size, mode='bicubic')
      return x

class DecodeHead(nn.Module):

  def __init__(self, embed_dim, num_classes):
      super().__init__()
      self.decode_head = BNHead(embed_dim=embed_dim, num_classes=num_classes)

  def forward(self, x, target_size):
      x = self.decode_head(x, target_size=target_size)
      return x

----------------------------------------------------------------------------------------------------
For my evaluation code, I achieved perfect 1.0 when evaluating ground_truth and ground_truth :
def compute_iou_per_class(gt: torch.Tensor, pred: torch.Tensor, num_classes: int):
    '''gt and pred can be any shape, 1 dim would be simple and easy'''
    ious = []
    for cls in range(num_classes):
        # Create boolean masks for the current class across the entire batch.
        gt_mask = (gt == cls)
        pred_mask = (pred == cls)
        
        # Compute intersection and union over batch, height, and width.
        intersection = torch.logical_and(gt_mask, pred_mask).sum().item()
        union = torch.logical_or(gt_mask, pred_mask).sum().item()
        
        if union == 0:
            ious.append(float('nan'))
        else:
            ious.append(intersection / union)
    return torch.tensor(ious)

----------------------------------------------------------------------------------------------------
They explictly mention they do not us TTA for linear method. The only suspecious thing could be I am not using mmcv script given in repo for inferencing cause serious version conflict that I can not fix. Does anyone know where is the problem? Maybe some magic trick in mmcv?

**Comment by JianwenCao** [comment]
It turned out to be I did not ignore the "cream color" edge and hard case label in VOC, the pretrained head is not trained in that class. All good now.

**Comment by davnords** [comment]
Hi @JianwenCao, would it be possible to share your entire code? It would be greatly appreciated. Especially if you skip the reliance on MMSeg :)

**Comment by JianwenCao** [comment]
Sorry the code is part of my lab project, it is private repo for now and I think I do not have the right to share entire code. But it should be easy, you initialize the dinov2 backbone model, after forwading the features, you plug the DecodeHead I give in the code. The mmseg is for normalization, you need to rewrite all the normalization method they mentioned in the config file.
